src/test/ui/suggest-remove-refs-2.stderr

```diff
@@ -0,0 +1,15 @@
+error[E0277]: the trait bound `&&&&&std::iter::Enumerate<std::slice::Iter<'_, {integer}>>: std::iter::Iterator` is not satisfied
+  --> $DIR/suggest-remove-refs-2.rs:14:19
+   |
+LL |     for (i, n) in & & & & &v.iter().enumerate() {
+   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |                   |
+   |                   `&&&&&std::iter::Enumerate<std::slice::Iter<'_, {integer}>>` is not an iterator; maybe try calling `.iter()` or a similar method
+   |                   help: consider removing 5 references `&`: `v.iter().enumerate()`
```

Comment:
After travis passes, I'm happy to merge, but I think we could tweak this wording slightly to make it clearer.

@nikomatsakis, @Mark-Simulacrum are you ok with this wording, or would you prefer something like

```
consider removing 5 `&`-references
```

or

```
consider removing the 5 leading `&`
```

Reply:
I think we definitely want the "references" in there, and I think we should remove `v.iter().enumerate()` from the help since it somewhat looks odd (since it's not what we're removing). Otherwise though I think this is good as-is.

Reply:
What about using `node_id` from `ObligationCause`? I am still searching only for occurrences of `&` in the snippet.
In case we don't need to use `node_id`, I should remove it from the codebase. Don't you think?

Reply:
@Mark-Simulacrum we could replace the `span_suggestion` with `span_suggestion_short` so that the code is not being shown in the label when displayed inline, the suggestion is displayed in full when it is shown on its own (like the multiline case above) and the tooling will be able to apply the correct output as it would now.

@ysiraichi would you mind doing that change? Once that is done, r=me.

Reply:
Sure... Which one?

Reply:
`span_suggestion` -> `span_suggestion_short` and

```
consider removing the 5 leading `&`-references
```

This line would end up looking like this:

```
LL |     for (i, n) in & & & & &v.iter().enumerate() {
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                   |
   |                   `&&&&&std::iter::Enumerate<std::slice::Iter<'_, {integer}>>` is not an iterator; maybe try calling `.iter()` or a similar method
   |                   help: consider removing the 5 leading `&`-references
```

If you had the extra time and felt it was worth it, it'd be great to change the suggestion's span to be

```
LL |     for (i, n) in & & & & &v.iter().enumerate() {
   |                   ---------^^^^^^^^^^^^^^^^^^^^^
   |                   |
   |                   `&&&&&std::iter::Enumerate<std::slice::Iter<'_, {integer}>>` is not an iterator; maybe try calling `.iter()` or a similar method
   |                   help: consider removing the 5 leading `&`-references
```

so that the suggestion will have a different color. That being said, I _won't_ block this PR on that.